packages/lib/slots.ts

```diff
@@ -86,23 +86,24 @@ function buildSlotsWithDateRanges({
     slotStartTime = slotStartTime.add(offsetStart ?? 0, "minutes").tz(timeZone);
 
     // Find the nearest appropriate slot boundary if this time falls within an existing slot
-    const slotBoundariesArray = Array.from(slotBoundaries.keys()).map((t) => dayjs(t));
-    if (slotBoundariesArray.length > 0) {
-      slotBoundariesArray.sort((a, b) => a.valueOf() - b.valueOf());
+    const slotBoundariesValueArray = Array.from(slotBoundaries.keys());
+    if (slotBoundariesValueArray.length > 0) {
+      slotBoundariesValueArray.sort((a, b) => a - b);
 
       let prevBoundary = null;
-      for (let i = slotBoundariesArray.length - 1; i >= 0; i--) {
-        if (slotBoundariesArray[i].isBefore(slotStartTime)) {
-          prevBoundary = slotBoundariesArray[i];
+      for (let i = slotBoundariesValueArray.length - 1; i >= 0; i--) {
+        if (slotStartTime.valueOf() >= slotBoundariesValueArray[i]) {
+          prevBoundary = slotBoundariesValueArray[i];
           break;
         }
       }
 
       if (prevBoundary) {
-        const prevBoundaryEnd = prevBoundary.add(frequency + (offsetStart ?? 0), "minutes");
+        const prevBoundaryEnd = dayjs(prevBoundary).add(frequency + (offsetStart ?? 0), "minutes");
```

Comment:
This is where we go back to dayjs to ensure the consumer doesn't notice this logic. Note that the instantiation just happens later, this is not passthrough from anywhere else.